doc/introduction/measurements.rst

```diff
@@ -153,6 +154,47 @@ so corresponds to a :math:`99.75\%` probability of measuring
 state :math:`|00\rangle`, and a :math:`0.25\%` probability of
 measuring state :math:`|01\rangle`.
 
+State
+-----
+
+Support for returning the quantum state of the QNode is also provided. Similar to the
+:func:`~.pennylane.probs` measurement function, **observables should not be input** into the
+:func:`~.state` function. Moreover, the state must be returned over **all** wires in the device
```

Comment:
This is obviously a bit strange from a user perspective: An argument that one has to set to a very particular value without any options. You mention that otherwise there were side effects, is it not better to deal with them?

Reply:
(I know this is usually easier said than done :)

Reply:
If the state is returned over all wires, why does the user need to even provide any arguments? :thinking:

Reply:
> we didn't require state() to have a wires argument, but this led to some side effects

Would any of the previous side effects be relevant here too?